src/OrchardCore/OrchardCore/Environment/Shell/ShellHost.cs

```diff
@@ -177,11 +177,11 @@ async Task CreateAndRegisterShellsAsync()
             else
             {
                 // Load all tenants, and activate their shell.
-                await Task.WhenAll(allSettings.Select(async settings =>
+                await Task.WhenAll(allSettings.Select(settings =>
                 {
                     try
                     {
-                        await GetOrCreateShellContextAsync(settings);
+                        return GetOrCreateShellContextAsync(settings);
```

Comment:
Hum, I'm tempted to think this will just hide the problem, as this code path will no longer catch "async exceptions" (e.g exceptions thrown after an `await` or using `Task.FromException`).

Reply:
I agree. I am reverting the change.

Reply:
Done. Please ensure this was not related to the issue this PR is fixing.

Reply:
The issue was fixed without this change that i did at the end, so no problem to revert it.
Sorry, i didn't see that it avoids to catch some exceptions.